r-cairo installation fails

Hi,

I am trying to install all R packages from spack in an environment. I can install most of them properly, but `r-cairo` and others dependent on this one fail.


Then I proceed to install packages and get the following for `r-cairo`:
```sh
spack -d install
...
  >> 147    xlib-backend.c:34:10: fatal error: X11/Intrinsic.h: No such file or directory
     148       34 | #include <X11/Intrinsic.h>      /*->    Xlib.h  Xutil.h Xresource.h .. */
     149          |          ^~~~~~~~~~~~~~~~~
     150    compilation terminated.
  >> 151    make: *** [xlib-backend.o] Error 1
     152    make: *** Waiting for unfinished jobs....
     153    ERROR: compilation failed for package 'Cairo'

...
```

### Information on my system

```sh
$ spack debug report
* **Spack:** 0.16.1-1621-e9c399110e
* **Python:** 3.6.8
* **Platform:** linux-centos7-cascadelake
* **Concretizer:** original
```
### Additional information
My environment config file looks like this:
```yaml
spack:
  # add package specs to the `specs` list
  #concretization: separately
  concretization: together
  packages:
    all:
      compiler:
      - gcc@10.2.0
      target:
      - linux-centos7-x86_64
    r:
      version:
      - 4.0.4
      variants: +X+rmath
    cairo:
      version:
      - 1.16.0
      variants: +X+fc+ft+gobject+pdf+png
    harfbuzz:
      version:
      - 2.6.8
      variants: +graphite2
    proj:
      version:
      - 7.2.1
  view: true
  specs:
  - r-...
  - ...
```

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have searched the issues of this repo and believe this is not a duplicate
- [x] I have run the failing commands in debug mode and reported the output

--- a/var/spack/repos/builtin/packages/r-cairo/package.py
+++ b/var/spack/repos/builtin/packages/r-cairo/package.py
@@ -31,6 +31,7 @@ class RCairo(RPackage):
     version('1.5-10', sha256='7837f0c384cd49bb3342cb39a916d7a80b02fffbf123913a58014e597f69b5d5')
     version('1.5-9', sha256='2a867b6cae96671d6bc3acf9334d6615dc01f6ecf1953a27cde8a43c724a38f4')
 
+    depends_on('r+X', type=('build', 'run'))
     depends_on('r@2.4.0:', type=('build', 'run'))
     depends_on('cairo@1.2:')
     depends_on('libxt')

--- a/var/spack/repos/builtin/packages/r-cairo/package.py
+++ b/var/spack/repos/builtin/packages/r-cairo/package.py
@@ -33,3 +33,4 @@ class RCairo(RPackage):
 
     depends_on('r@2.4.0:', type=('build', 'run'))
     depends_on('cairo@1.2:')
+    depends_on('libxt')

